CorsixTH/Lua/utility.lua

```diff
@@ -311,3 +311,14 @@ function rangeMapLookup(number, buckets)
   end
   assert(false) -- Should never get here.
 end
+
+-- work around for bitwise operators
+-- see http://lua-users.org/wiki/BitwiseOperators
+function bit(p)
+  return 2 ^ (p - 1)  -- 1-based indexing
+end
+
+-- Typical call:  if hasbit(x, bit(3)) then ...
```

Comment:
This pattern is actually the only sane way to call this function, why not fold the "bit" function into this one, and make the second parameter the bit number you want to test?
(oh, and please don't make this 1-based, preserving the 2^p relation is a lot saner imho.)

Also, document the function. See also https://github.com/CorsixTH/CorsixTH/wiki/CodeDocumentationStyle